### PR TITLE
Change color in insert mode

### PR DESCRIPTION
**Pull request by MaraniMatias**
Hello, good morning, thanks for your work, I like it.

Add a new functionality,
When you enter the insert mode, change the color of the first block of the status line.

- Normal mode

image

- Insert mode

image

### CODE REVIEW

**Comment by liuchengxu** [comment]
Thanks for your work. Though this tiny plugin is not planned to be sophisticated, I'm fine with your changes, which look nice. However, before I merge your PR, please remain the origin color of `eleline.vim` part and the first block in normal mode, only change the first block's color in insert mode.

File: plugin/eleline.vim [hunk 1]
```diff
@@ -191,6 +193,20 @@ function! s:hi_statusline()
...
+  if a:mode == 'i'
+    call s:hi('User1' , 239 ,  179 )
+  elseif a:mode == 'r'
+    call s:hi('User1' , 239 ,  196 )
+  else
+    call s:hi('User1' , 7   , s:bg+8 )
+  endif
+endfunction
+
+au InsertEnter * call InsertStatuslineColor(v:insertmode)
```
Comment (liuchengxu) [inline plugin/eleline.vim]: Wrap these `au` in `augroup eleline` below, see https://www.reddit.com/r/vim/wiki/vimrctips#wiki_wrap_your_autocmds_in_proper_augroups.

**Comment by MaraniMatias** [comment]
I will correct it

File: plugin/eleline.vim [hunk 2]
```diff
@@ -191,6 +192,16 @@ function! s:hi_statusline()
   call s:hi('User9'      , 251 , s:bg+5 )
 endfunction
 
+function! InsertStatuslineColor(mode)
+  if a:mode == 'i'
+    call s:hi('User1' , 232 ,  171 )
```
Comment (liuchengxu) [inline plugin/eleline.vim]: Hello, I prefer your previous color change in insert mode,  pale grey is better , something like this:

```vim
    call s:hi('User1' , 251 ,  s:bg+8 )
```

FYI, at least not the pink one :). What do you think?

**Comment by liuchengxu** [comment]
hmm, what I mean is to swap your current two color changes https://github.com/liuchengxu/eleline.vim/pull/10/commits/6d9762b927348d1d1362b24babe559005d4165c0:

- remain all the colors in normal mode as before
- only change the first block's color to pale grey in insert mode


```diff
@@ -173,7 +173,7 @@ if has('termguicolors') && &termguicolors
 endif

 function! s:hi_statusline()
-  call s:hi('User1'      , 251 , s:bg+8 )
+  call s:hi('User1'      , 232 , 178 )
   call s:hi('paste'      , 232 , 178    , 'bold')
   call s:hi('User2'      , 178 , s:bg+8 )
   call s:hi('User3'      , 250 , s:bg+6 )
@@ -194,7 +194,7 @@ endfunction

 function! InsertStatuslineColor(mode)
   if a:mode == 'i'
-    call s:hi('User1' , 232 , 178  )
+    call s:hi('User1' , 251 , s:bg+8  )
   elseif a:mode == 'r'
     call s:hi('User1' , 232 ,  160 )
   else
@@ -223,7 +223,7 @@ augroup eleline
   " Change colors for insert mode
   autocmd InsertEnter * call InsertStatuslineColor(v:insertmode)
   autocmd InsertChange * call InsertStatuslineColor(v:insertmode)
-  autocmd InsertLeave * call s:hi('User1', 251 , s:bg+8 )
+  autocmd InsertLeave * call s:hi('User1', 232 , 178 )
 augroup END

 let &cpoptions = s:save_cpo
```

**Comment by MaraniMatias** [comment]
I'm sorry, I didn't understand you, my English is not very good.

image

image

**Comment by liuchengxu** [comment]
Never mind, I'm also not a native English speaker :).

Merged, thanks! 🍻

**Pull request merged**